### PR TITLE
Using ransack for search

### PR DESCRIPTION
**Pull request by thonght-3051**
## Related Tickets
- [#53419](https://edu-redmine.sun-asterisk.vn/issues/53419)

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
image

### CODE REVIEW

**Comment by thonght-3051** [comment]
ready

File: app/models/product.rb [hunk 1]
```diff
@@ -30,4 +36,19 @@ class Product < ApplicationRecord
   def sum_quantity
     product_attributes.sum(:quantity)
   end
+
+  # def self.ransackable_attributes(auth_object = nil)
+  #   # binding.pry
+  #   if auth_object == :admin
+  #     %w(name)
+  #   end
+  # end
```
Comment (huongnt-2545) [inline app/models/product.rb]: ko dùng đến thì xóa đi em nhé

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@
 # Use this hook to configure devise mailer, warden hooks and so forth.
 # Many of these configuration options can be set straight in your model.
 Devise.setup do |config|
-  config.omniauth :google_oauth2, ENV["google_client_id"],
+  config.omniauth :google_oauth2, ENV["google_client_id"],,
```
Comment (huongnt-2545) [inline config/initializers/devise.rb]:
```suggestion
  config.omniauth :google_oauth2, ENV["google_client_id"],
```

File: db/seeds.rb [hunk 3]
```diff
@@ -72,10 +72,10 @@
   )
 }
 
-Product.all.each { |pro| pro.product_images.create!(
-  image: Faker::LoremFlickr.image
-  )
-}
+# Product.all.each { |pro| pro.product_images.create!(
```
Comment (huongnt-2545) [inline db/seeds.rb]: chạy xong thì bỏ comment đi em

**Comment by thonght-3051** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
Cập nhật ticker redmine

**Pull request merged**